dotnet/src/SemanticKernel.Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Web;
+
+/// <summary>
+/// Skill to download web files.
+/// </summary>
+public class WebFileDownloadSkill : IDisposable
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Where to save file.
+        /// </summary>
+        public const string FilePath = "filePath";
+    }
+
+    private readonly ILogger _logger;
+    private readonly HttpClientHandler _httpClientHandler;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Constructor for WebFileDownloadSkill.
+    /// </summary>
+    /// <param name="logger">Optional logger.</param>
+    public WebFileDownloadSkill(ILogger<WebFileDownloadSkill>? logger = null)
+    {
+        this._logger = logger ?? NullLogger<WebFileDownloadSkill>.Instance;
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new HttpClient(this._httpClientHandler);
+    }
+
+    [SKFunction("Downloads a file to local storage")]
+    [SKFunctionInput(Description = "URL of file to download")]
+    [SKFunctionContextParameter(Name = Parameters.FilePath, Description = "Path where to save file locally")]
+    public async Task DownloadToFileAsync(string uri, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(DownloadToFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.FilePath, out string filePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(DownloadToFileAsync)}");
+            string errorMessage = $"Missing variable {Parameters.FilePath}";
+            context.Fail(errorMessage);
+
+            throw new KeyNotFoundException(errorMessage);
+        }
+
+        this._logger.LogDebug("Sending GET request for {0}", uri);
+        HttpResponseMessage response = await this._httpClient.GetAsync(new Uri(uri), context.CancellationToken);
+        response.EnsureSuccessStatusCode();
+        this._logger.LogDebug("Response received: {0}", response.StatusCode);
+
+        using Stream webStream = await response.Content.ReadAsStreamAsync();
+        using FileStream outputFileStream = new FileStream(filePath, FileMode.Create);
```

Comment:
So here I'm just using regular file methods (that is, not SK file system skills).  Does this make sense in this context?

Reply:
I think it's fine since the FileIOSkill is about interacting with file contents, not the file as a whole.